python/composio/tools/toolset.py

```diff
@@ -998,11 +1040,33 @@ def create_integration(
 
     def initiate_connection(
         self,
-        integration_id: str,
+        integration_id: t.Optional[str] = None,
+        app: t.Optional[AppType] = None,
         entity_id: t.Optional[str] = None,
         redirect_url: t.Optional[str] = None,
         connected_account_params: t.Optional[t.Dict] = None,
     ) -> ConnectionRequestModel:
+        if integration_id is None and app is None:
+            raise ComposioSDKError(
+                message="Both `integration_id` and `app` cannot be None"
+            )
+
+        if integration_id is None:
+            try:
+                integration_id = (
+                    self.get_entity(id=entity_id or self.entity_id)
+                    .get_connection(app=app)
+                    .integrationId
+                )
+            except NoItemsFound as e:
+                raise ComposioSDKError(
```

Comment:
In the `initiate_connection` method, the error handling for `NoItemsFound` could be improved. Consider adding more specific error information, such as the app name or entity ID, to help with debugging.